client/app/dbaas/logs/detail/streams/logs-streams.service.js

```diff
@@ -0,0 +1,128 @@
+class LogsStreamsService {
+    constructor ($q, OvhApiDbaas, ServiceHelper) {
+        this.$q = $q;
+        this.ServiceHelper = ServiceHelper;
+
+        this.LogsApiService = {
+            all: OvhApiDbaas.Logs().Lexi()
+        };
+
+        this.StreamsApiService = {
+            all: OvhApiDbaas.Logs().Stream().Lexi()
+        };
+
+        this.AccountingAapiService = {
+            all: OvhApiDbaas.Logs().Accounting().Aapi()
+        };
+    }
+
+    /**
+     * returns array of streams with details of logged in user
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to array of streams. each stream will have all details populated.
+     * @memberof LogsStreamsService
+     */
+    getStreams (serviceName) {
+        return this.getStreamDetails(serviceName)
+            .then(streams => streams.map(stream => this.transformStream(serviceName, stream)))
+            .catch(this.ServiceHelper.errorHandler("logs_streams_get_error"));
+    }
+
+    /**
+     * gets stream details for each stream in array
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to array of streams
+     * @memberof LogsStreamsService
+     */
+    getStreamDetails (serviceName) {
+        return this.getAllStreams(serviceName)
+            .then(streams => {
+                const promises = streams.map(stream =>
+                    this.getStream(serviceName, stream));
+                return this.$q.all(promises);
+            });
+    }
+
+    /**
+     * returns details of a stream
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @returns promise which will be resolve to stream object
+     * @memberof LogsStreamsService
+     */
+    getStream (serviceName, streamId) {
+        return this.StreamsApiService.all.get({
+            serviceName,
+            streamId
+        })
+            .$promise
+            .then(stream => stream);
+    }
+
+    /**
+     * returns array of stream id's of logged in user
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to array of stream id's
+     * @memberof LogsStreamsService
+     */
+    getAllStreams (serviceName) {
+        return this.LogsApiService.all.streams({ serviceName })
+            .$promise;
```

Comment:
No error handling?  It might be ok if this function is a kind of private function

Reply:
its a private method